zipline-cli/src/main/kotlin/app/cash/zipline/cli/ValidateZiplineApi.kt

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (C) 2023 Cash App
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package app.cash.zipline.cli
+
+import app.cash.zipline.api.validator.ActualApiHasProblems
+import app.cash.zipline.api.validator.ExpectedApiIsUpToDate
+import app.cash.zipline.api.validator.ExpectedApiRequiresUpdates
+import app.cash.zipline.api.validator.fir.readFirZiplineApi
+import app.cash.zipline.api.validator.makeApiCompatibilityDecision
+import app.cash.zipline.api.validator.toml.TomlZiplineApi
+import app.cash.zipline.api.validator.toml.readTomlZiplineApi
+import app.cash.zipline.api.validator.toml.writeTomlZiplineApi
+import com.github.ajalt.clikt.core.CliktCommand
+import com.github.ajalt.clikt.core.CliktError
+import com.github.ajalt.clikt.parameters.options.convert
+import com.github.ajalt.clikt.parameters.options.help
+import com.github.ajalt.clikt.parameters.options.option
+import com.github.ajalt.clikt.parameters.options.required
+import com.github.ajalt.clikt.parameters.types.path
+import java.io.File
+import kotlin.io.path.exists
+import okio.buffer
+import okio.sink
+import okio.source
+
+/**
+ * Compares a Kotlin compilation's actual declared APIs against expected APIs and reports potential
+ * problems.
+ *
+ * Actual APIs are extracted by running the frontend of the Kotlin compiler (FIR).
+ *
+ * Expected APIs are persisted as TOML files.
+ *
+ * Which command is executed decides what happens when the actual API (.kt files) declares more
+ * services or functions than expected (.toml file):
+ *
+ *  * zipline-api-check fails the build.
+ *  * zipline-api-dump updates the TOML file.
+ *
+ * Both modes fail the build if the converse is true; ie. the actual API declares fewer services
+ * or functions than expected.
+ *
+ * Both modes succeed if the actual APIs are equal to the expectations.
+ */
+class ValidateZiplineApi(
+  name: String,
+) : CliktCommand(name = name) {
+  private val tomlFile by option("--toml-file")
+    .path()
+    .required()
+    .help("Path to the TOML file that this command will read and write")
+
+  private val sources by option("--sources")
+    .convert { it.split(File.pathSeparator).map(::File) }
+    .required()
+
+  private val classpath by option("--class-path")
+    .convert { it.split(File.pathSeparator).map(::File) }
+    .required()
```

Comment:
Clikt API is so so nice I love it. Thank you, @ajalt!

Reply:
Thanks for the kind words! 

If you didn't know about the built-in [split][1], you could use it to make this even a little simpler:

```kotlin
val classpath: List<File> by option("--class-path")
    .file().split(File.pathSeparator).required()
```


[1]: https://ajalt.github.io/clikt/api/clikt/com.github.ajalt.clikt.parameters.options/split.html